src/modules/UserModule.cpp

```diff
@@ -0,0 +1,295 @@
+	}
+
+	doc.setObject(docObj);
+}
+
+void UserModule::OnLoad(const QJsonDocument& doc)
+{
+	QJsonObject docObj = doc.object();
+	QStringList servers = docObj.keys();
+
+	userDescriptions.clear();
+
+	// User data
+	for (auto server : servers)
+	{
+		snowflake_t guildId = server.toULongLong();
+		QJsonObject obj = docObj[server].toObject();
+		QStringList users = obj.keys();
+		QList<UserDescription> descriptions;
+
+		for (const QString& user : users)
+		{
+			auto userObj = obj[user].toObject();
+			
+			UserDescription description {
+				user.toULongLong(),
+				userObj["name"].toString(),
+				userObj["location"].toString(),
+				userObj["industry"].toString(),
+				userObj["programmingInterests"].toString(),
+				userObj["currentlyWorkingOn"].toString(),
+				userObj["githubLink"].toString(),
+			};
+
+			descriptions.append(description);
+		}
+
+		userDescriptions.insert(guildId, descriptions);
+	}
+}
+
+void UserModule::OnMessage(Discord::Client& client, const Discord::Message& message)
+{
+	client.getChannel(message.channelId()).then(
+		[this, message, &client](const Discord::Channel& channel)
+		{
+			snowflake_t guildId = channel.guildId();
+			
+			if (guildId == 0 || message.author().bot())
+			{
+				// It's a bot / we're in a DM
+				return;
+			}
+
+			DescriptionData& descriptionData = guildDescriptionData[guildId];
+			snowflake_t authorId = message.author().id();
+			QString messageContent = message.content();
+			
+			if (descriptionData.messageId == message.id())
+			{
+				// This is the !iam command, ignore it
+				return;
+			}
+
+			if (descriptionData.isBeingUsed && descriptionData.userId == authorId)
+			{
+				QString contentToWrite = messageContent;
+				bool shouldWrite = true;
+				
+				if (messageContent == "skip")
+				{
+					contentToWrite = "";
+				}
+				else if (messageContent == "continue")
+				{
+					shouldWrite = false;
+				}
+				else if (messageContent == "cancel")
+				{
+					*descriptionData.currentUserDescription = descriptionData.oldUserDescription;
+					descriptionData.isBeingUsed = false;
+					delete descriptionData.timer;
+
+					client.createMessage(message.channelId(), "**Cancelled setting description\n**Reverting to old values.");
+					
+					return;
+				}
+
+				if (shouldWrite)
+				{
+					switch (descriptionData.questionUpTo)
+					{
+					case 0:
+						descriptionData.currentUserDescription->name = contentToWrite;
+						break;
+					case 1:
+						descriptionData.currentUserDescription->location = contentToWrite;
+						break;
+					case 2:
+						descriptionData.currentUserDescription->industry = contentToWrite;
+						break;
+					case 3:
+						descriptionData.currentUserDescription->programmingInterests = contentToWrite;
+						break;
+					case 4:
+						descriptionData.currentUserDescription->currentlyWorkingOn = contentToWrite;
+						break;
+					case 5:
+						descriptionData.currentUserDescription->githubLink = contentToWrite;
+						break;
+					}
+				}
```

Comment:
Is there some way you can make it easier to add more questions in the future? So you only have to add one line, or at least only change one place of the code base?

Reply:
To elaborate: This switch maps to the question array in the header file. Can this connection be changed so maybe it's a list of pairs with questions strings paired with a `QVariant` or lambda?